cpp/src/Slice/Scanner.l

```diff
@@ -46,11 +47,13 @@ namespace
 
     void initScanner();
     void preAction();
+    void yynoreturn fatalError(const char* msg);
```

Comment:
I assume yynoreturn is a macro?

Should we just use https://en.cppreference.com/w/cpp/language/attributes/noreturn for readability?

Reply:
Yes, it's a macro. This is the definition provided by `flex`:
```
#if defined(__GNUC__) && __GNUC__ >= 3
#    define yynoreturn __attribute__((__noreturn__))
#else
#    define yynoreturn
#endif
```
So, it's not a simple drop-in replacement for `[[noreturn]]`, since it's platform dependent...

For the record, I didn't just find this macro. It's from the signature of the default-generated error message function.

Maybe we could put the attribute on, and it's fine if the signature is slightly different then what flex expects (on some platforms), but I'd rather play it safe. This stuff is already an error-prone black box.